### ISSUE TITLE
Include instructions on honor code page

### ISSUE THREAD

**Issue by jonatanschroeder** [PrairieLearn/PrairieLearn#3087]
The honor code page includes a very simple statement that students need to pledge. However, different courses, different exams and different contexts define what is allowed and what is not allowed differently. It would be good to have the ability to include some extra information in this page, maybe an HTML specific to the assessment, that includes information about rules for what's allowed or not allowed, also potentially including the topics covered by the assessment.

**Comment by armandofox** [comment]
Yes, in general i doubt you will be able to write an honor code that satisfies everyone, so it' d be great if the `infoAssessment.json` could specify (eg) the name of a static html file (maybe in `clientFilesCourse`?) that can specify this language, possibly with a way to interpolate the student's name, course number, etc. (using `data[]`?) , that the student would be required to acknowledge and click thru before doing the assessment.  I thought about just adding an `honorCode` attribute to `infoAssessment.json`, but it would be nice to be able to do simple formatting like bold, bullet list, etc. so an HTML file is better.

**Comment by jonatanschroeder** [comment]
I like the idea of a file, though it would need to be in a location that is covered by chunks (see https://github.com/PrairieLearn/PrairieLearn/pull/4029#issuecomment-812969310). `clientFilesCourse` (or `serverFilesCourse` is an interesting approach, since it would give the advantage of allowing the same pattern to be used in multiple assessments, though I'm not sure if this makes sense from an organization point of view.

As for adding the student's name, etc. this could be achieved with moustache.

**Comment by eliotwrobson** [comment]
This seems like it could be a use case for the proposed `pl-template` element #4701, since you're really just populating an honor code template provided by the instructor. See https://github.com/PrairieLearn/PrairieLearn/pull/6747

**Comment by nwalters512** [comment]
@jonatanschroeder I have a long-standing PR that adds assessment chunks: https://github.com/PrairieLearn/PrairieLearn/pull/4031. I really need to get that merged...

I'd much prefer referencing files in `serverFiles*` than implementing full-blown PrairieLearn element rendering on this page. We can still support data interpolation via Mustache, as Jonatan points out above.

**Comment by nwalters512** [comment]
Calling out that I don't want arbitrary HTML on this page. I'd be OK with heavily-sanitized Markdown. Mustache/EJS interpolation is almost certainly overkill.

**Comment by jonatanschroeder** [comment]
> * We'll process the string as Markdown and convert it to HTML. We'll be careful to only permit heavily sanitized Markdown.

My minor concern about this is that `text` currently accepts HTML, so it would become inconsistent. If sanitization is expected, we have been using DOMPurify in our elements to take care of scripts, etc. and ensure that HTML has properly closed tags, etc.. You can also have the editor in the settings page use quill, which has been working well in the pl-rich-text-editor element.

> * We'll process the string with Mustache so that `{{name}}` can resolve to the student's name.

Again, `text` uses EJS. While I prefer Mustache, we might want to consider changing `text` to support Mustache as well for consistency, with some option for EJS backwards compatibility (e.g., check for `<%`). I don't feel strongly about this one, but it's probably something to discuss.

**Comment by nwalters512** [comment]
> My minor concern about this is that text currently accepts HTML, so it would become inconsistent.

IMO it was a mistake for `text` to accept arbitrary HTML. Yes, things will become inconsistent for now; the end goal is Markdown everywhere.

Likewise, `text` should indeed support Mustache.

**Comment by jonatanschroeder** [comment]
> IMO it was a mistake for `text` to accept arbitrary HTML. Yes, things will become inconsistent for now; the end goal is Markdown everywhere.

I don't want to push back too much, but: it would be nice to have a WYSIWYG editor for these fields in the assessment editor (e.g., something like the pl-rich-text-editor). If this is a future goal, using HTML here would be ideal, since converting that to Markdown would be pretty hard, as previous experience has demonstrated. However if you believe this is not desirable I understand and respect the use of Markdown.

I should point out that DOMPurify (which we already use in several elements) has the ability to sanitize HTML [with a strict allowlist set of tags and attributes](https://github.com/cure53/DOMPurify?tab=readme-ov-file#control-our-allow-lists-and-block-lists), which would likely take care of your concern with sanitizing this input.

Again, I'm ok with your push towards Markdown if you believe that's the direction we want to go, I just want to make sure we consider the alternatives.
